include/bitcoin/bitcoin/utility/resource_lock.hpp

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2011-2015 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * libbitcoin is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License with
+ * additional permissions to the one published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version. For more information see LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef LIBBITCOIN_FILE_LOCK_HPP
+#define LIBBITCOIN_FILE_LOCK_HPP
+
+#include <boost/filesystem.hpp>
+#include <boost/interprocess/sync/file_lock.hpp>
+
+namespace libbitcoin {
+
+typedef boost::filesystem::path file_path;
```

Comment:
Avoid making this a global bc typedef. If necessary, define in the context of this class (as "path", by existing convention) but preferably just use the fully qualified name in the header and use a local typedef in th cpp file.